### PR TITLE
Added PDO support

### PR DESCRIPTION
**Pull request by bosoniq**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -453,6 +453,38 @@ $populator->addEntity('Book', 5, array(), array(
...
+
+Faker provides an adapter for using a PHP Data Object (PDO) to populate one or more database tables.
+
+To populate a table create a new populator class (using a generator and PDO instance as parameters), then run the `addRows()` method providing a data mapping for your desired table, the number of rows you wish to insert and the table you wish to insert them into.
+
+```php
+<?php
+$generator = \Faker\Factory::create();
+$populator = new Faker\ORM\PDO\Populator($generator, $pdo);
+$populator->addRows($usersMapping, 5, 'users');
```
Comment (fzaninotto) [inline readme.md]: Why don't you use the same syntax as for ORM populators?

File: src/Faker/ORM/PDO/Populator.php [hunk 2]
```diff
@@ -0,0 +1,173 @@
+<?php
```
Comment (fzaninotto) [inline src/Faker/ORM/PDO/Populator.php]: missing unit tests

File: src/Faker/ORM/PDO/Populator.php [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+     */
+    public function addRows($mapping, $number, $table)
+    {
+        if (!isset($this->rows[$table])) {
+            $this->rows[$table] = array();
+        }
+
+        for ($i=0; $i < $number; $i++) {
+            $temp = array();
+            foreach ($mapping as $key => &$value) {
```
Comment (fzaninotto) [inline src/Faker/ORM/PDO/Populator.php]: strange `&` here, can you explain why you use it?

Comment (fzaninotto) [reply src/Faker/ORM/PDO/Populator.php]: rename `$value` to a more meaningful variable name

File: readme.md [hunk 4]
```diff
@@ -453,6 +453,38 @@ $populator->addEntity('Book', 5, array(), array(
...
+$populator->addRows($recordsMapping, 10, 'records');
+$insertedPKs = $populator->execute();
+```
+
+You can insert rows into multiple tables in one execution by simply using the `addRows()` method for all the tables you want to write to whilst supplying a valid mapping for each.
+
```
Comment (fzaninotto) [inline readme.md]: the ORM populators also handle fkeys, you should support it, too.

File: src/Faker/ORM/PDO/Populator.php [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+
+    /**
+     * Convert the rows corresponding to the supplied table name to
+     * keys for use in pdo execute statement.
+     * @param  string $tableKey The table name this dataset is for.
+     * @return array            The array with placeholder keys
+     */
+    private function convertToPlaceholders($tableKey)
+    {
+        $originalArray = $this->rows[$tableKey];
```
Comment (fzaninotto) [inline src/Faker/ORM/PDO/Populator.php]: please use more meaningful variable names. `$originalArray`, `$key`, `$value` tell me the type of the variable, not what they contain.

File: src/Faker/ORM/PDO/Populator.php [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+
+        return $newArray;
+    }
+
+    /**
+     * Return a faker value from the correct formatter.
+     * @param  string/array $request Data from supplied mapping
+     * @return string                Value returned by formatter or null
+     */
+    private function generateValue($request)
```
Comment (fzaninotto) [inline src/Faker/ORM/PDO/Populator.php]: `$request` doesn't mean anything here. Use the other populators as an example of naming.

File: src/Faker/ORM/PDO/Populator.php [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+
+            if ($count === 1) {
+                $params = array();
+            } else {
+                $params = $request[1];
+            }
+
+            // Run method
+            switch (count($params)) {
+                case 0:
```
Comment (fzaninotto) [inline src/Faker/ORM/PDO/Populator.php]: There must be a better way of doing it, using `call_user_func` or something.

File: src/Faker/ORM/PDO/Populator.php [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+     * @param  string/array $request Data from supplied mapping
+     * @return string                Value returned by formatter or null
+     */
+    private function generateValue($request)
+    {
+        $final = false;
+        $reqType = gettype($request);
+        $count = count($request);
+
+        if ($reqType === 'array' && $count > 0) {
```
Comment (fzaninotto) [inline src/Faker/ORM/PDO/Populator.php]: split this method into two, one for string, the other for array.

File: src/Faker/ORM/PDO/Populator.php [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+                    break;
+                case 5:
+                    $final = $this->generator->$name($params[0], $params[1], $params[1], $params[3], $params[4]);
+                    break;
+                case 6:
+                    $final = $this->generator->$name($params[0], $params[1], $params[1], $params[3], $params[4], $params[5]);
+                    break;
+            }
+        } else if ($reqType == 'string') {
+            $name = trim($request);
```
Comment (fzaninotto) [inline src/Faker/ORM/PDO/Populator.php]: what's `$name`? Impossible to tell by reading your code... Please find a better name.

File: src/Faker/ORM/PDO/Populator.php [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+            $first = strpos($name, '(');
```
Comment (fzaninotto) [inline src/Faker/ORM/PDO/Populator.php]: what's `$first`?

**Comment by pimjansen** [comment]
@bosoniq any update on this since the PR is stale. Thanks!

**Comment by pimjansen** [comment]
I am going to close this PR as stale. Please feel free to reopen if needed

**Pull request closed without merging**